File: js/boj/15649.js
//백준 15649 n과m1

const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split(' ').map(nm => parseInt(nm));

// const input = require('fs').readFileSync('input.txt').toString().trim().split(' ');
// const [n, m] = [4, 4]
const n = input.shift();
const m = input.shift();

let result = '';

const visited = new Array(10).fill(false);
const arr = [];

function dfs(k) {
    if (k===m) {
        result += `${arr.join(' ')}\n`;
        return;
    }
    for (let i=1; i<n+1; i++) {
        if (visited[i]===true) continue;
        visited[i] = true;
        arr.push(i);
        dfs(k+1);
        arr.pop();
        visited[i] = false;
    }
}

dfs(0);

console.log(result.trim())

File: 15649.py
#boj 15649 n과 m 1

N, M = map(int, input().split())
v = [0] * (N+1)

def dfs(n, lst):
    global N, M, answer, v
    if n==M:
        print(*lst)
        return
    for j in range(1, N+1):
        if v[j] == 0:
            lst.append(j)
            v[j] = 1
            dfs(n+1, lst)
            v[j] = 0
            lst.pop()

dfs(0, [])